COBOL Programming Course #2 - Advanced Topics/COBOL Programming Course #2 - Advanced Topics.md

```diff
@@ -665,6 +665,512 @@ In a multithreaded environment, there are some limitations on COBOL programs. In
 To see more details on the limitation of COBOL with multithreading, check out the [Programming Guide](https://www.ibm.com/docs/en/cobol-zos/6.3?topic=multithreading-handling-cobol-limitations).
 
 \newpage
+
+# SORT and MERGE
+Sometime when we want to process a file we need it to be sorted, COBOL provides the `SORT` verb for this.
+
+If we have two or more file that are sorted using the same key or keys, we can combine them into one sorted file
+using COBOL `MERGE` verb.
+
+- **SORT**
+    - **The formats of the SORT statement**
+    - **INPUT PROCEDURE phrase**
+    - **DUPLICATES phrase**
+        - **Format 1**
+        - **Format 2**
+    - **ASCENDING KEY and DESCENDING KEY phrases (format 2)**
+    - **SORT Example: Sorting an employee data file**
+- **Merge**
+    - **The Merge statement Format**
+    - **Merge Example: merging employee, and interns data files together**
+- **SORT & MERGE**
+    - **USING phrase**
+    - **GIVING phrase**
+    - **OUTPUT PROCEDURE phrase**
+    - **ASCENDING/DESCENDING KEY phrase**
+        - **KEYs rules**
+        - **KEYs types**
+        - **COLLATING SEQUENCE phrase**
+        - **Segmentation considerations**
+
+
+## SORT
+The `SORT` statement causes a set of records or table elements to be arranged in a user-specified sequence.
+
+For sorting **files**, the `SORT` statement accepts records from one or more files, sorts them according to the specified
+keys, and makes the sorted records available either through an output procedure or in an output file.
+
+For sorting **tables**, the `SORT` statement sorts table elements according to specified table keys.
+
+### The formats of the SORT statement
+The `SORT` statement has two formats,
+the first format is not supported for programs that are compiled with the `THREAD`
+option, while the second can be used with programs that are compiled with the `THREAD` option.
+
+both formats can be used anywhere in the `PROCEDURE DIVISION`.
+
+#### **Format 1 for sorting files:**
+![Sort statement format](Images/sort-and-merge/SORT-format-1.png)
+_Format 1 of a sort statement_
+
+`file-name-1`:
+- the sort process uses this file as temporary file during the sort
+- The name given in the SD entry that describes the records to be sorted.
```

Comment:
Ahmed, just to make things more consistent with the rest of the course, I would suggest reformatting the bullet points in the three chapters into paragraph :)

Reply:
sure, will read through the chapters and reorganize them into paragraphs 👍